File: Appium/Native/Python/ios/native.py
from appium.webdriver.common.mobileby import MobileBy
from selenium import webdriver
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.ui import WebDriverWait

 #  1. Replace <<cloud name>> with your perfecto cloud name (e.g. demo is the cloudName of demo.perfectomobile.com).
cloudName = "<<cloud name>>"

capabilities = {
    #  2. Replace <<security token>> with your perfecto security token.
    'securityToken' : "<<security token>>",
    
    
    # 3. Set device capabilities.
    'platformName': 'iOS',
    'manufacturer': 'Apple',
    'model': 'iPhone.*',
    
     # 4. Set Perfecto Media repository path of App under test.
    'app': 'PUBLIC:ExpenseTracker/Native/iOS/InvoiceApp1.0.ipa',
    
    # 5. Set the unique identifier of your app
    'bundleId': 'io.perfecto.expense.tracker',
    
    # Set other capabilities.
    'enableAppiumBehavior': True, # Enable new architecture of Appium
    'autoLaunch': True, # To work with hybrid applications, install the iOS/Android application as instrumented.
    'iOSResign': True, # Resigns using developer certificate
    # 'fullReset': false, # Reset app state by uninstalling app
    'takesScreenshot': False,
    'screenshotOnError': True,
    'openDeviceTimeout': 5
}
# Initialize the Appium driver
driver = webdriver.Remote('https://' + cloudName + '.perfectomobile.com/nexperience/perfectomobile/wd/hub', capabilities)
# set implicit wait time 
driver.implicitly_wait(5)
# Your code goes here

wait = WebDriverWait(driver, 30)


email = wait.until(EC.presence_of_element_located((MobileBy.NAME, "login_email")))
email.send_keys('test@perfecto.com')

password = wait.until(EC.presence_of_element_located((MobileBy.NAME, "login_password")))
password.send_keys('test123')

login = wait.until(EC.presence_of_element_located((MobileBy.NAME, "login_login_btn")))
login.click()

add = wait.until(EC.presence_of_element_located((MobileBy.NAME, "list_add_btn")))
add.click()

head = wait.until(EC.presence_of_element_located((MobileBy.NAME, "edit_head")))
head.click()

flight = wait.until(EC.presence_of_element_located((MobileBy.XPATH, "//*[@value='- Select -']")))
flight.send_keys("Flight") 

amount = wait.until(EC.presence_of_element_located((MobileBy.NAME, "edit_amount")))
amount.send_keys('100')

save = wait.until(EC.presence_of_element_located((MobileBy.NAME, "add_save_btn")))
save.click()

expectedText = "Please enter valid category"
wait.until(EC.presence_of_element_located((MobileBy.XPATH, "//*[@name='"+expectedText+"']")))

#Quits the driver
driver.quit()
print("Python iOS Execution completed")

File: Appium/Native/Ruby/ios/native.rb
require 'perfecto-reporting'
require 'appium_lib'

desired_caps = {
        #  1. Replace <<cloud name>> with your perfecto cloud name (e.g. demo is the cloudName of demo.perfectomobile.com).
    appium_lib: {
        server_url: 'https://%s.perfectomobile.com/nexperience/perfectomobile/wd/hub' % "<<cloud name>>", 
    },
    caps: {
        #  2. Replace <<security token>> with your perfecto security token.
        securityToken: '<<security token>>',
        
          # 3. Set device capabilities.
          platformName: 'iOS',
          manufacturer: 'Apple',
          model: 'iPhone.*',
  
          # 4. Set Perfecto Media repository path of App under test.
          app: 'PUBLIC:ExpenseTracker/Native/iOS/InvoiceApp1.0.ipa',
          
          # 5. Set the unique identifier of your app
          bundleId: 'io.perfecto.expense.tracker',
  
          # Set other capabilities.
          enableAppiumBehavior: true, # Enable new architecture of Appium
          autoLaunch: true, # Whether to have Appium install and launch the app automatically.
          iOSResign: true, # Resign with developer certificate
          # fullReset: false, # Whether to install and launch the app automatically.
          takesScreenshot: false,
          screenshotOnError: true,
          openDeviceTimeout: 5
        }
}
# Initialize the Appium driver
@driver = Appium::Driver.new(desired_caps, true).start_driver
# Setting implicit wait
@driver.manage.timeouts.implicit_wait = 5

# Initialize Smart Reporting
if ENV["jobName"] != nil
    perfectoExecutionContext = PerfectoExecutionContext.new(PerfectoExecutionContext::PerfectoExecutionContextBuilder
    .withWebDriver(@driver).withJob(Job.new(ENV["jobName"], ENV["jobNumber"].to_i)).build)
else
    perfectoExecutionContext = PerfectoExecutionContext.new(PerfectoExecutionContext::PerfectoExecutionContextBuilder
            .withWebDriver(@driver).build)
end
@reportiumClient = PerfectoReportiumClient.new(perfectoExecutionContext)
tec = TestContext::TestContextBuilder.build()
@reportiumClient.testStart("Native Ruby iOS Sample", tec)

begin
    wait = Selenium::WebDriver::Wait.new(:timeout => 30)

    @reportiumClient.stepStart('Enter email')
    wait.until{ @driver.find_element(:name => 'login_email') }
    @driver.find_element(:name => 'login_email').send_keys('test@perfecto.com')
    @reportiumClient.stepEnd()

    @reportiumClient.stepStart('Enter password')
    wait.until{ @driver.find_element(:name => 'login_password').displayed? }
    @driver.find_element(:name => 'login_password').send_keys('test123')
    @reportiumClient.stepEnd()

    @reportiumClient.stepStart('Click login')
    @driver.hide_keyboard
    wait.until{ @driver.find_element(:name => 'login_login_btn').displayed? }
    @driver.find_element(:name => 'login_login_btn').click
    @reportiumClient.stepEnd()

    @reportiumClient.stepStart('Add expense')
    wait.until{ @driver.find_element(:name => 'list_add_btn').displayed? }
    @driver.find_element(:name => 'list_add_btn').click
    @reportiumClient.stepEnd()

    @reportiumClient.stepStart('Select head')
    wait.until{ @driver.find_element(:name => 'edit_head').displayed? }
    @driver.find_element(:name => 'edit_head').click
    wait.until{ @driver.find_element(:xpath => "//*[@value='- Select -']").displayed? }
    @driver.find_element(:xpath => "//*[@value='- Select -']").send_keys("Flight") 
    @reportiumClient.stepEnd()

    @reportiumClient.stepStart('Enter amount')
    wait.until{ @driver.find_element(:name => 'edit_amount').displayed? }
    @driver.find_element(:name => 'edit_amount').send_keys('100') 
    @reportiumClient.stepEnd()

    @reportiumClient.stepStart('Save expense')
    wait.until{ @driver.find_element(:name => 'add_save_btn').displayed? }
    @driver.find_element(:name => 'add_save_btn').click
    @reportiumClient.stepEnd()

    @reportiumClient.stepStart("Verify alert");
    expectedText = "Please enter valid category"
    wait.until{ @driver.find_element(:name => expectedText).displayed? }
    @reportiumClient.reportiumAssert(expectedText, @driver.find_element(:name => expectedText).text === expectedText)
    @reportiumClient.stepEnd();
            
    @reportiumClient.testStop(TestResultFactory.createSuccess(), tec)

rescue Exception => exception
    @exception = exception
    @reportiumClient.testStop(TestResultFactory.createFailure(@exception.exception.message, @exception.exception, nil), tec)
    raise exception
ensure
    # Prints the report url
     puts 'Report url - ' + @reportiumClient.getReportUrl
     
    #Quits the driver
    @driver.quit
    puts "Ruby iOS Execution completed"
end